common/primitives/src/handles.rs

```diff
@@ -0,0 +1,79 @@
+use crate::msa::MessageSourceId;
+#[cfg(feature = "std")]
+use crate::utils::*;
+use codec::{Decode, Encode};
+use frame_support::BoundedVec;
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+use sp_core::ConstU32;
+use sp_std::vec::Vec;
+
+/// The minimum base handle (not including suffix or delimiter) length in characters
+pub const HANDLE_BASE_CHARS_MIN: u32 = 3;
+/// The minimum base handle (not including suffix or delimiter) length in bytes
+pub const HANDLE_BASE_BYTES_MIN: u32 = 1 * HANDLE_BASE_CHARS_MIN;
+/// The maximum base handle (not including suffix or delimiter) length in characters
+pub const HANDLE_BASE_CHARS_MAX: u32 = 20;
+/// The maximum base handle (not including suffix or delimiter) length in bytes
+pub const HANDLE_BASE_BYTES_MAX: u32 = 32; // Hard limit of 32 bytes
+/// The maximum number of digits in a suffix
+pub const SUFFIX_MAX_DIGITS: usize = 5; // The max value of a HandleSuffix (u16) is 65535 which is 5 digits.
+/// The maximum count of suffixes allowed to be requested at once
+pub const MAX_SUFFIXES_COUNT: u16 = 100;
+/// The default count of suffixes to request if none is provided
+pub const DEFAULT_SUFFIX_COUNT: u16 = 1;
+/// A handle (base, canonical, or display)
+pub type Handle = BoundedVec<u8, ConstU32<HANDLE_BASE_BYTES_MAX>>;
```

Comment:
I think we can move this back into the handles lib since we do not use it outside that scope.

Reply:
Either or can do, not picky about it but do not have a local types.rs in pallet